### PR TITLE
production/GFSv16: frozen precip density fix

### PR DESCRIPTION
**Pull request by HelinWei-NOAA**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsibility to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

The snow density is used for all frozen precip types in the current operational GFS. With this assumption the density for sleet or freezing rain is too small and the snow depth is overestimated in this situation. This fix will give the different density to snow, graupel, and freezing rain in the Noah LSM driver.

The fix will change the result.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes https://github.com/ufs-community/ufs-weather-model/issues/1268

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [ ] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs

- waiting on noaa-emc/fv3atm/pull/[546](https://github.com/NOAA-EMC/fv3atm/pull/546)

### CODE REVIEW

**Comment by HelinWei-NOAA** [comment]
Is this conflict from the fact that I created my fv3 branch before the commit of "Bug fix for ceiling height in GFS V16 in-line post"?

**Comment by junwang-noaa** [comment]
@HelinWei-NOAA Yes. Please merge your ufs-weather-model branch and fv3atm branches with the latest production/GFS.v16 branches.

**Review by yangfanglin — APPROVED** [review]

**Review by yangfanglin — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by junwang-noaa** [comment]
@HelinWei-NOAA Would you please create an issue to describe the problem and solution in your PR? Thanks

**Comment by HelinWei-NOAA** [comment]
> @HelinWei-NOAA Would you please create an issue to describe the problem and solution in your PR? Thanks

Done. [issue 1268](https://github.com/ufs-community/ufs-weather-model/issues/1268)

**Comment by HelinWei-NOAA** [comment]
> @HelinWei-NOAA Yes. Please merge your ufs-weather-model branch and fv3atm branches with the latest production/GFS.v16 branches.

Done

**Comment by junwang-noaa** [comment]
@HelinWei-NOAA Please update the .gitmodules to point to your fv3atm branch with the fix:

cd ufs-weather-model
git branch (should see your noaalsm_snowflx  branch)
vi .gitmodules (update fv3atm to point to your fv3 branch noahlsm_snowfix)
git submodule sync
cd FV3
git remote -v( should see your branch)
git remote update
git checkout noahlsm_snowfix
git submodule update --init --recursive
cd ..
git status (should fv3 .gitmodule and FV3 updated)
git commit -a -m"update fv3atm branch"
git push origin your_ufs-weather-model branch

**Comment by HelinWei-NOAA** [comment]
> @HelinWei-NOAA Please update the .gitmodules to point to your fv3atm branch with the fix:
> 
> cd ufs-weather-model git branch (should see your noaalsm_snowflx branch) vi .gitmodules (update fv3atm to point to your fv3 branch noahlsm_snowfix) git submodule sync cd FV3 git remote -v( should see your branch) git remote update git checkout noahlsm_snowfix git submodule update --init --recursive cd .. git status (should fv3 .gitmodule and FV3 updated) git commit -a -m"update fv3atm branch" git push origin your_ufs-weather-model branch

Done. Thanks for the instructions.

**Comment by junwang-noaa** [comment]
@HelinWei-NOAA Would you please point to fv3atm production/GFS.v16 branch (revision: [9a9a701](https://github.com/NOAA-EMC/fv3atm/commit/9a9a701333305131fa7fd20ed32b8c3cf8570399) )? Thanks

**Comment by HelinWei-NOAA** [comment]
done
> @HelinWei-NOAA Would you please point to fv3atm production/GFS.v16 branch (revision: [9a9a701](https://github.com/NOAA-EMC/fv3atm/commit/9a9a701333305131fa7fd20ed32b8c3cf8570399) )? Thanks

File: .gitmodules [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [submodule "FV3"]
 	path = FV3
 	url = https://github.com/NOAA-EMC/fv3atm
-	branch = production/GFS.v16
+	branch = GFS.v16
```
Comment (junwang-noaa) [inline .gitmodules]: @HelinWei-NOAA The branch is: production/GFS.v16. Please update.

Comment (HelinWei-NOAA) [reply .gitmodules]: Done

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**